lte/gateway/c/core/oai/test/amf/test_amf_stateless.cpp

```diff
@@ -0,0 +1,295 @@
+      0xff, 0x91, 0xa1, 0x28, 0x01, 0x00, 0x7b, 0x00, 0x07, 0x80, 0x00,
+      0x0a, 0x00, 0x00, 0x0d, 0x00, 0x12, 0x01, 0x81, 0x22, 0x01, 0x01,
+      0x25, 0x09, 0x08, 0x69, 0x6e, 0x74, 0x65, 0x72, 0x6e, 0x65, 0x74};
+
+  const uint8_t pdu_sess_release_hexbuf[14] = {0x7e, 0x00, 0x67, 0x01, 0x00,
+                                               0x06, 0x2e, 0x01, 0x01, 0xd1,
+                                               0x59, 0x24, 0x12, 0x01};
+
+  const uint8_t pdu_sess_release_complete_hexbuf[12] = {
+      0x7e, 0x00, 0x67, 0x01, 0x00, 0x04, 0x2e, 0x01, 0x01, 0xd4, 0x12, 0x01};
+
+  uint8_t ue_initiated_dereg_hexbuf[24] = {
+      0x7e, 0x01, 0x41, 0x21, 0xe6, 0xe2, 0x03, 0x7e, 0x00, 0x45, 0x01, 0x00,
+      0x0b, 0xf2, 0x22, 0x62, 0x54, 0x01, 0x00, 0x40, 0x0,  0x0,  0x0,  0x0};
+};
+
+TEST_F(AMFAppStatelessTest, TestStateless) {
+  int rc                 = RETURNerror;
+  amf_ue_ngap_id_t ue_id = 0;
+
+  /* Send the initial UE message */
+  imsi64_t imsi64 = 0;
+  imsi64          = send_initial_ue_message_no_tmsi(
+      amf_app_desc_p, 36, 1, 1, 0, plmn, initial_ue_message_hexbuf,
+      sizeof(initial_ue_message_hexbuf));
+  AMFClientServicer::getInstance().map_tableKey_protoStr.clear();
+  EXPECT_TRUE(AMFClientServicer::getInstance().map_tableKey_protoStr.isEmpty());
+  // Writes the state to the data store
+  put_amf_nas_state();
+  EXPECT_FALSE(
+      AMFClientServicer::getInstance().map_tableKey_protoStr.isEmpty());
+
+  /* Check if UE Context is created with correct imsi */
+  EXPECT_TRUE(get_ue_id_from_imsi(amf_app_desc_p, imsi64, &ue_id));
+
+  /* Send the authentication response message from subscriberdb */
+  rc = send_proc_authentication_info_answer(imsi, ue_id, true);
+  EXPECT_EQ(rc, RETURNok);
+
+  /* Send uplink nas message for auth response from UE */
+  rc = send_uplink_nas_message_ue_auth_response(
+      amf_app_desc_p, ue_id, plmn, ue_auth_response_hexbuf,
+      sizeof(ue_auth_response_hexbuf));
+  EXPECT_EQ(rc, RETURNok);
+
+  /* Send uplink nas message for security mode complete response from UE */
+  rc = send_uplink_nas_message_ue_smc_response(
+      amf_app_desc_p, ue_id, plmn, ue_smc_response_hexbuf,
+      sizeof(ue_smc_response_hexbuf));
+  EXPECT_EQ(rc, RETURNok);
+
+  send_initial_context_response(amf_app_desc_p, ue_id);
+
+  /* Send uplink nas message for registration complete response from UE */
+  rc = send_uplink_nas_registration_complete(
+      amf_app_desc_p, ue_id, plmn, ue_registration_complete_hexbuf,
+      sizeof(ue_registration_complete_hexbuf));
+  EXPECT_EQ(rc, RETURNok);
+
+  // Calling TearDown() and SetUp() simulates a service restart.
+  // Clears the state
+  AMFAppStatelessTest::TearDown();
+  EXPECT_TRUE(amf_app_desc_p->amf_ue_contexts.imsi_amf_ue_id_htbl.isEmpty());
+  EXPECT_TRUE(amf_app_desc_p->amf_ue_contexts.tun11_ue_context_htbl.isEmpty());
+  EXPECT_TRUE(amf_app_desc_p->amf_ue_contexts.guti_ue_context_htbl.isEmpty());
```

Comment:
Can we also include a check that amf_ngap_ue_id_ue_context_htbl is empty here?

Reply:
In this, we are syncing only amf_app_desc whereas ue_context is not synced. The ue_context syncing and amf_ue_ngap_id_ue_context_htbl check is added as part of the Additional unit test cases PR( #11034).